### ISSUE TITLE
Trin and latest versions should shoul install each other

### ISSUE THREAD

**Issue by ml019** [hamlet-io/executor-python#206]
## Expected Behaviour
If I install the latest `train` engine, then the corresponding versioned release should also show an installed and vica-versa.

## Current Behaviour
The train and its corresponding versioned engine must be installed separately

## Possible Solution
When installing an engine, check if any other engine has the same sha and make it as installed as well.

## Context
Usability - it seems odd to have to install the two separately - unless there is some difference between them?

**Comment by roleyfoley** [comment]
The difference would be when it comes to updates 

An explicit version will not update 
A named version will update 

Each engine has its own file system section so keeping them separate isolates the engines from getting in the way of each other.

What would you expect as the behaviour here?

**Comment by ml019** [comment]
I guess the question is "If I install the explicit version that is currently the train, should my local train move with it? " Currently until I explicit ask for the latest train, I am actually running behind. 

Similarly, if I update the train and it has moved to the next versioned release, the new version will still need to be explicitly installed.

I guess we can run with it and see if the existing behaviour is preferrable - I just found on new install that after I had updated the train, it still showed the latest version as needing to be installed which felt wrong.

Basically are "train" and "v8.2.1" refs to the same thing (like refs in git) or separate things.
